keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -472,6 +479,23 @@ public DoubleTensor logInPlace() {
         return this;
     }
 
+    /**
+     * This is identical to log().times(y), except that it changes NaN results to 0.
+     * This is important when calculating 0log0, which should return 0
+     * See https://arcsecond.wordpress.com/2009/03/19/0log0-0-for-real/ for some mathematical justification
+     * @param y The tensor value to multiply by
+     * @return the log of this tensor multiplied by y
+     */
+    @Override
+    public DoubleTensor logTimesInPlace(DoubleTensor y) {
```

Comment:
Awesome. I like the idea of putting a word in the method name that denotes it's doing validation as otherwise it may appear to be a shortcut. Perhaps: `safeLogTimes` or `validatedLogTimes` or `nanFreeSince93LogTimes`. (Not really on the last one)

Reply:
`safeLogTimes` sounds good to me